src/core/src/main/java/org/locationtech/geogig/data/EPSGBoundsCalc.java

```diff
@@ -0,0 +1,125 @@
+
+        if (null == domainOfValidity) {
+            throw new Exception("No domain of validity provided by CRS definition");
+        }
+
+        Collection<? extends GeographicExtent> geographicElements;
+        geographicElements = domainOfValidity.getGeographicElements();
+
+        if (null == geographicElements || geographicElements.size() != 1) {
+            throw new Exception("Number of geographic elements != 1");
+        }
+
+        GeographicExtent geographicExtent = geographicElements.iterator().next();
+        if (!(geographicExtent instanceof GeographicBoundingBox)) {
+            throw new Exception("geographic extent is not a geographic bounding box");
+        }
+
+        if (!geographicExtent.getInclusion()) {
+            throw new Exception("geographic extent is exclusive, can only deal with inclusive ones");
+        }
+
+        GeographicBoundingBox geographicBoundingBox = (GeographicBoundingBox) geographicExtent;
+
+        double minx = geographicBoundingBox.getWestBoundLongitude();
+        double miny = geographicBoundingBox.getSouthBoundLatitude();
+        double maxx = geographicBoundingBox.getEastBoundLongitude();
+        double maxy = geographicBoundingBox.getNorthBoundLatitude();
+
+        CoordinateReferenceSystem wgs84LongFirst;
+
+        try {
+
+            wgs84LongFirst = CRS.decode("EPSG:4326", true);
+            CoordinateOperationFactory coordOpFactory = CRS.getCoordinateOperationFactory(true);
+            CoordinateOperation op = coordOpFactory.createOperation(wgs84LongFirst,crs);
+
+            ReferencedEnvelope refEnvelope = new ReferencedEnvelope(minx, maxx, miny, maxy, wgs84LongFirst);
+            GeneralEnvelope genEnvelope = CRS.transform(op, refEnvelope);
+
+            double xmax = genEnvelope.getMaximum(0);
+            double ymax = genEnvelope.getMaximum(1);
+            double xmin = genEnvelope.getMinimum(0);
+            double ymin = genEnvelope.getMinimum(1);
+
+            Envelope envelope = new Envelope(xmin, xmax, ymin, ymax);
+            return envelope;
+        } catch (Exception e) {
+            throw new Exception("ERROR: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Search for the given CRS (EPSG code), return the bounds (domain of validity)
+     * @param refId the input CRS
+     * @return projectionBounds an Envelope describing the CRS bounds, or null if bounds are not found
+     */
+    public Envelope findCode(String refId) throws Exception {
+        Envelope projectionBounds = null;
+        CoordinateReferenceSystem crs = null;
+
+        CRSAuthorityFactory authorityFactory = CRS.getAuthorityFactory(true);
+        Set<String> authorityCodes = authorityFactory.getAuthorityCodes(CoordinateReferenceSystem.class);
+
+        if (refId.contains("WGS 84")) {
+            crs = authorityFactory.createCoordinateReferenceSystem("EPSG:4326");
+            projectionBounds = getExtents(crs);
```

Comment:
Might want to just return the -180/-90 to 180/90 directly here - why are you special casing this?